CMakeLists.txt

```diff
@@ -10,6 +10,36 @@ if(NOT DEFINED CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 11)
 endif()
 
+option(WITH_STL "Whether to use Standard Library for C++latest features" OFF)
+
+option(WITH_ABSEIL "Whether to use Abseil for C++latest features" OFF)
+
```

Comment:
@jsuereth - This is complicated, but I'll try to elaborate :)

- technically Abseil would be best for vs2015 support, because there is no alternate good variant working with vs2015. Only `absl::variant` works. The one we borrowed from MPark - `nostd::variant` - does not compile for vs2015, to be more exact - it has problems with complex aggregate variant types.

- now Abseil actually requires statically linked library of itself, which I 'hacked' - if we only use Abseil for variant and nothing else. Whereas GSL itself is header-only. In certain environments I'd prefer the full set of everything to be built with Standard Library and, preferably, header-only --- not needing Abseil (e.g. if I compile the OpenTelemetry SDK for inclusion in some Azure cloud Service).

So - it's problematic on all accounts everywhere:
- `nostd::variant` - the one we have today, is not compiling on vs2015. The only one compiling is `absl::variant`
- `absl::variant` - if used without any hacks, requires Abseil STATIC lib (messy)
- whereas GSL only has `gsl::span` and does not have `gsl::variant` :(

So -- to answer your question, in my setup I would prefer:
- just a portion of Abseil for `absl::variant` (but not the entire Abseil) + MS-GSL for `span` ... whereas
- Google projects would probably prefer Abseil for everything ... whereas
- dynamic libs prefer `nostd::` classes... except this won't work for vs2015. We may need to borrow just the `variant` from Abseil, tweak it so that it doesn't require static linking of the core Abseil, and replace MPark (`nostd::variant`) with Abseil... Which I wanted to do in a separate PR.